source/distributed/shared_tria.cc

```diff
@@ -76,14 +146,84 @@ namespace parallel
                   active_halo_layer.find(cell) == active_halo_layer.end())
                 cell->set_subdomain_id(numbers::artificial_subdomain_id);
             }
+
+          // loop over all cells in multigrid hierarchy and mark artificial:
+          if (settings & construct_multigrid_hierarchy)
+            {
+              std_cxx11::function<bool (const typename parallel::shared::Triangulation<dim,spacedim>::cell_iterator &)> predicate
```

Comment:
@davydden I think that the flag <tt>construct_multigrid_hierarchy</tt> is still useful for a computation where the mesh is distributed manually, we only don't want to call the function <tt>GridTools::partition_multigrid_levels</tt> as the user must manual distribute the level cells. 

For example here we would like to be able to use the flag  <tt>construct_multigrid_hierarchy</tt> alongside <tt>partition_custom_signal</tt> to mark artificial cells.